### ISSUE TITLE
add 'name' option to createDynamicVat

### ISSUE THREAD

**Issue by warner** [Agoric/agoric-sdk#1325]
**What is the Problem Being Solved?**
@Chris-Hibbert was debugging some zoe/zcf code this morning, and mentioned that it would be nice if we could give the kernel a name for each dynamic vat, which could be displayed in error messages (especially if/when that vat terminates due to a metering underflow).

In #1307 I added an options bag, to support non-metered dynamic vats. I can easily add a `name` to the set of options.

**Description of the Design**
Instead of `E(vatService).createVat(bundle)`, Zoe can do `E(vatService).createVat(bundle, { name: `zcf for ${contractName}` })`. Kernel debug messages that reference the dynamic vat (in particular creation and termination) should include the name.

Static vats have a name (which comes from keys on the `.vats` property of the config object passed into `makeVatController`, and is frequently derived from a filename like `vat-zoe.js` when the config object is built with `scanBasedir`). These names appear in the `vats` argument to the bootstrap message, and thus must be unique. The uniqueness is maintained by virtue of coming from *keys* of `config.vats`.

All vats (static and dynamic) have an ID, which is a `v` prefixed to a sequential integer.

We might want to use `description` rather than `name`. The caller of `createVat()` is under no obligation to come up with unique names. It doesn't matter quite as much for dynamic vats, because there's nothing that needs to enumerate them (they don't go into the `vats` argument of the bootstrap method, which is long gone by the time dynamic vats are created). But still, if `name` doesn't have the same properties for static and dynamic vats, it might be better to use a different identifier.

**Security Considerations**
What code, if any, might rely upon this identifier?

**Test Plan**
I intend to augment the tests in `SwingSet/test/vat-admin/` to exercise calls to `createVat` both with and without the description, and make sure nothing breaks. I'll manually inspect the output to see if the description is present in the appropriate debug messages.

The kernel error/debug messages that would reference it are not readily available to unit tests, so I don't plan to automate checking them (they get sent to the console, and we aren't trying to guarantee anything about what gets printed there, so capturing `console.log` calls and inspecting them would make the tests brittle against subsequent changes).

**Comment by warner** [comment]
Also, at least for now, we should log the creation of dynamic vats, just like we do for the creation of static ones. The message should announce the vatID number so later messages (like "kernel panic" or vat termination) which don't (yet?) include the description can be correlated.

**Comment by Chris-Hibbert** [comment]
I'll add this here, so we don't forget. When Zoe/Zcf creates a new vat, it should include a count as part of the dynamic name so we can trace different actions across publicAuction#3 compared to publicAuction#12.  I think the count would be associated with the installation.

**Comment by warner** [comment]
Do we have a table where contract installations get names? If we add something to a registry somewhere, that would be a good string to feed into the logs (the part before the counter).

When we get to the point where contract bundles are stored in a large-string store, and indexed by their hash, we can use the hash as a strong name, and then maybe the logger will have its own petname table to assign a more meaningful name to each hash.

**Comment by katelynsills** [comment]
> Do we have a table where contract installations get names? If we add something to a registry somewhere, that would be a good string to feed into the logs (the part before the counter).

Nope, installations are currently nameless within Zoe. While they can be made public by being put in the `board`, the board only gives a numeric ID.

**Comment by warner** [comment]
This should probably be more of a description than a name, and it should not be confusable with the (unique) names assigned to static vats by the config file.
